citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/AuthorityService.java

```diff
@@ -201,6 +203,14 @@ public IAuthorityEntry create(IAuthorityEntry entry, IUser user) {
         return save(entry);
     }
 
+    @Override
+    @Transactional
+    public IAuthorityEntry create(IAuthorityEntry entry, IUser user, String uri) {
+        entry = create(entry, user);
+        entry.setUri(uri + entry.getId());
+        return save(entry);
```

Comment:
have the other create method call this one and move the code from the other create method here; otherwise you'll end up with two save operations which is unnecessary.

Reply:
I can merge both methods but wouldn't we require to save once to generate an Id so that we can append it to the generated URI. Also, we need to distinguish between the case where we need to generate a URI and where we don't. So, I created another method.

Reply:
jup, you're right. let's leave it the way it is but make a comment.

Reply:
okay

Reply:
comment is still missing

Reply:
I changed the method name, which I think is self explanatory. I'll add the comment for better clarity.

Reply:
the comment should be before or after `create` was called saying the it needs to be persisted first before we can get its id to create the uri.